### PR TITLE
Refactor edit endpoints

### PR DESCRIPTION
**Pull request by McGarvs**
Does not move into utils yet, will do that in another pull request.

### CODE REVIEW

**Comment by McGarvs** [comment]
To test editTestPlan, send the following in a put request:
{
  "_id": "6172500699915be1b771acb3", 
  "name": "LTE Failover",
  "detailedDescription": "Test failover to LTE during upstream failure",
  "version": "1234",
  "deviceConfig": "[{'system':{'system_id':'LTE Router'}}, []]",
  "customerFeedback": "Works as expected",
  "testCases": ["619003fec01efe48e4b3fe0d"],
  "authors": [""],
  "isActive": false,
  "summaryBOM": [
        {
            "isOptional": true,
            "quantity": 12,
            "deviceId": "617251a399915be1b771acb6"
        }
    ],
  "createdOn": "2021-10-21T07:00:00.000Z"
}

**Comment by McGarvs** [comment]
To test editEngagement, send the following in a put request:
{
  "_id": "617249b199915be1b771aca1",
  "name":"Engagement 2",
  "statusCode": 2,
  "engagementDetails": "Test Details 2",
  "SE": "1234",
  "POC_Engineer": "2345",
  "customer": "Small Finance",
  "SFDC": "https://cradlepoint.lightning.force.com/lightning/r/Opportunity/006380...",
  "testPlanId": "61724e5599915be1b771acb2",
  "createdOn": "2021-11-20T15:00:00.000Z"
}

**Comment by McGarvs** [comment]
To test editTest, send the following in a put request:
{
  "_id": "6192e77bb769d28dff5942c9", 
  "details": "test post details",
  "name": "test name",
  "results": ["617252cd99915be1b771acbe"],
  "testCaseId": "618ff52599912cea5510f06b"
}

**Comment by McGarvs** [comment]
To test editTestCase, send the following in a post request:
{
    "_id": "618ff93999912cea5510f07d",
    "timeEstimate": 1.3,
    "description": "This is a description.",
    "config": "This is a config.",
    "name": "Test Case 2",
    "tests": [
        "6192e77bb769d28dff5942c9"
    ],
    "BOM": [
        {
            "isOptional": true,
            "quantity": 12,
            "deviceId": "617251a399915be1b771acb6"
        }
    ],
    "testPlanId": "61724e5599915be1b771acb2"
}

**Comment by McGarvs** [comment]
To test editResult, send the following in a put request:
{
  "_id": "61930925b09972144203dd2c", 
  "details": "test detail",
  "POCApproval": "0",
  "SEApproval": "1",
  "evidence": "test evidence",
  "testId": "619007a699912cea5510f098"
}

File: cradlepoint/pages/api/editTestPlan.js [hunk 1]
```diff
@@ -13,15 +13,21 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await testPlanSchema.isValid(data);
     if (valid){
-      const result = testPlanSchema.cast(data);
+      const validData = testPlanSchema.cast(data);
+      console.log("\nvalidData:", validData);
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTestPlan.js]: remove console.log

File: cradlepoint/pages/api/editTestPlan.js [hunk 2]
```diff
@@ -13,15 +13,21 @@ export default async (req, res) => {
...
+      const id = ObjectId(validData._id);
+      const summaryBOM = validData.summaryBOM.map(device => {
+        return {...device, deviceId: ObjectId(device.deviceId)}
+      });
+      const testCases = validData.testCases.map(testCaseId => ObjectId(testCaseId));
+      // Create the database query and replacement object
       const query = {_id: id};
-      const testPlan = {...result, summaryBOM, _id: id };
+      const newtestPlan = {...validData, summaryBOM:summaryBOM, testCases:testCases, _id: id };
+      console.log("\ntest:", newtestPlan);
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTestPlan.js]: remove console.log

File: cradlepoint/pages/api/editEngagement.js [hunk 3]
```diff
@@ -13,15 +13,16 @@ export default async (req, res) => {
...
+      const id = ObjectId(validData._id);
+      const testplanId = ObjectId(validData.testPlanId);
+      // Create the database query and replacement object
       const query = {_id: id};
-      const engagement = {...result , _id: id, testPlanId:testplanId};
+      const newEngagement = {...validData , _id: id, testPlanId:testplanId};
       // Update the Database w/ new Engagement
       const db = await connectToDb();
-      await db.collection("engagements").replaceOne(query, engagement);
+      await db.collection("engagements").replaceOne(query, newEngagement);
```
Comment (mtran-12) [inline cradlepoint/pages/api/editEngagement.js:25]: This endpoint isn't working for me. Changing just the objectId, I get input not in right format

File: cradlepoint/pages/api/editTest.js [hunk 4]
```diff
@@ -13,20 +13,21 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await testSchema.isValid(data);
     if (valid && ObjectId.isValid(data.testCaseId)){
-      const result = testSchema.cast(data);
+      const validData = testSchema.cast(data);
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTest.js:16]: Also says input not in right format

**Comment by mtran-12** [comment]
Only test plan endpoint worked for me
Supplied data did not validate for anything else

File: cradlepoint/pages/api/editTest.js [hunk 5]
```diff
@@ -13,20 +13,21 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await testSchema.isValid(data);
     if (valid && ObjectId.isValid(data.testCaseId)){
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTest.js:15]: shouldn't this check if the id for the result is valid as well?

File: cradlepoint/pages/api/editTestCase.js [hunk 6]
```diff
@@ -13,15 +13,20 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await testCaseSchema.isValid(data);
     if (valid && ObjectId.isValid(data.testPlanId)){
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTestCase.js:15]: Shouldn't this also check for valid test case id?

File: cradlepoint/pages/api/editTestPlan.js [hunk 7]
```diff
@@ -13,15 +13,19 @@ export default async (req, res) => {
     const data = req.body;
     const valid = await testPlanSchema.isValid(data);
     if (valid){
```
Comment (mtran-12) [inline cradlepoint/pages/api/editTestPlan.js:15]: shouldn't we check for valid id

File: cradlepoint/schemas/testPlanSchema.js [hunk 8]
```diff
@@ -7,10 +7,20 @@ export const testPlanSchema = yup.object().shape({
   version: yup.string().required(),
   deviceConfig: yup.string(),
   customerFeedback: yup.string(),
-  testCases: yup.array().required(),
+  // testCases: yup.array().required(),
```
Comment (mtran-12) [inline cradlepoint/schemas/testPlanSchema.js:10]: Comment should also be removed

**Review by mtran-12 — CHANGES_REQUESTED** [review]

File: cradlepoint/schemas/testPlanSchema.js [hunk 9]
```diff
@@ -7,10 +7,20 @@ export const testPlanSchema = yup.object().shape({
...
+  testCases: yup.array().of(
+    yup.string().required()
+  ).required(),
   authors: yup.array().required(),
   isActive: yup.bool().required(),
-  summaryBOM: yup.array().required(),
+  // summaryBOM: yup.array().required(),
```
Comment (mtran-12) [inline cradlepoint/schemas/testPlanSchema.js:16]: Comment should also be removed

File: cradlepoint/schemas/testPlanSchema.js [hunk 10]
```diff
@@ -7,10 +7,20 @@ export const testPlanSchema = yup.object().shape({
...
+  summaryBOM: yup.array().of(
+    yup.object().shape({
+      isOptional: yup.boolean().required(),
+      quantity: yup.number().positive().required(),
+      deviceId: yup.string().required(),
+    })
+  ).required(),
```
Comment (mtran-12) [inline cradlepoint/schemas/testPlanSchema.js:23]: Also need to add a check for a valid engagementId

File: cradlepoint/pages/api/editTestPlan.js (hunk 7 above)
Comment (mtran-12) [reply cradlepoint/pages/api/editTestPlan.js:15]: and for valid engagementIt

File: cradlepoint/pages/api/editEngagement.js (hunk 3 above)
Comment (McGarvs) [reply cradlepoint/pages/api/editEngagement.js:25]: Tried exactly that, and it worked for me!

File: cradlepoint/pages/api/editTest.js (hunk 4 above)
Comment (McGarvs) [reply cradlepoint/pages/api/editTest.js:16]: Works for me!

**Review by mtran-12 — APPROVED** [review]

**Pull request merged**